common/static/common/js/discussion/utils.js

```diff
@@ -151,61 +153,59 @@
-                    "   <button class='close-modal dismiss' title='" + gettext("Close") + "'>" +
-                    "       <span class='icon fa fa-times' aria-hidden='true'></span>" +
-                    "   </button>" +
-                    "   <header><h2/><hr/></header>" +
-                    "   <p id='discussion-alert-message'/><hr/>" +
-                    "   <button class='dismiss'>" + gettext("OK") + "</button>" +
-                    "</div>"
+                edx.HtmlUtils.append(
+                    $alertDiv,
+                    edx.HtmlUtils.joinHtml(
+                        edx.HtmlUtils.HTML("<div class='inner-wrapper discussion-alert-wrapper'>"),
+                            edx.HtmlUtils.interpolateHtml(
+                                edx.HtmlUtils.HTML("<button class='close-modal dismiss' title='{title}'>"),
+                                {title: gettext("Close")}
+                            ),
+                                edx.HtmlUtils.HTML("<span class='icon fa fa-times' aria-hidden='true'></span>"),
+                            edx.HtmlUtils.HTML("</button>"),
+                            edx.HtmlUtils.HTML("<header><h2/><hr/></header>"),
+                            edx.HtmlUtils.HTML("<p id='discussion-alert-message'/><hr/>"),
+                            edx.HtmlUtils.HTML("<button class='dismiss'>"),
+                                gettext("OK"),
+                            edx.HtmlUtils.HTML("</button>"),
+                        edx.HtmlUtils.HTML("</div>")
+                    )
                 );
-                this.makeFocusTrap(alertDiv.find("button"));
-                alertTrigger = $("<a href='#discussion-alert' id='discussion-alert-trigger'/>").css("display", "none");
-                alertTrigger.leanModal({
+                this.makeFocusTrap($alertDiv.find("button"));
+                $alertTrigger = $("<a href='#discussion-alert' id='discussion-alert-trigger'/>").css("display", "none");
+                $alertTrigger.leanModal({
                     closeButton: "#discussion-alert .dismiss",
                     overlay: 1,
                     top: 200
                 });
-                $("body").append(alertDiv).append(alertTrigger);
+                $("body").append($alertDiv).append($alertTrigger);
             }
-            $("#discussion-alert header h2").html(header);
-            $("#discussion-alert p").html(body);
+            $("#discussion-alert header h2").text(header);
+            $("#discussion-alert p").text(body);
             $("#discussion-alert-trigger").click();
-            return $("#discussion-alert button").focus();
+            $("#discussion-alert button").focus();
         };
 
         DiscussionUtil.safeAjax = function(params) {
             var $elem, deferred, request,
                 self = this;
             $elem = params.$elem;
-            if ($elem && $elem.attr("disabled")) {
+            if ($elem && $elem.prop("disabled")) {
```

Comment:
@bjacobel both. I recall there was some weird edge case when it didn't work as expected, but what was that exactly I don't remember. It might have something to do with 3rd party LMS one solutions client are using.